### DISCUSSION TITLE
CORS blocking PATCH API call

### DISCUSSION THREAD

**Discussion by mhdaouas** [directus/directus#9904] [Archive]
I'm unable to request a PATCH API because of a CORS problem. I get the following error:

> Access to XMLHttpRequest at 'http://127.0.0.1:8055/items/...' from origin 'http://127.0.0.1:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

What is the proper CORS config to resolve this issue?

**Comment by mhdaouas** [comment]
Up

**Reply by iqraaswad** [reply]
I think this is your patch api problem .. u should add cors policy to your api since directus itself and your api have different origin (different port number)

**Reply by mhdaouas** [reply]
I added this in the Directus .env file:
```
CORS_ENABLED=true
CORS_ORIGIN=http://127.0.0.1:8080
CORS_METHODS=GET,POST,PATCH,DELETE,OPTIONS
CORS_ALLOWED_HEADERS=Content-Type,Authorization,X-Requested-With
CORS_EXPOSED_HEADERS=Content-Range
CORS_CREDENTIALS=true
CORS_MAX_AGE=1800
```
Should I add another CORS configuration from front side ?

**Reply by iqraaswad** [reply]
that should works..

**Reply by mhdaouas** [reply]
No it didn't, that's the problem.

**Reply by rijkvanzanten** [reply]
Could be the browser preventing it still, even with the correct headers. I've had some issues before where browsers required the use of HTTPS in this case. Chromium based browsers often give some more information. Any other info in the console or network tab?
